File: VB/05_Tables/ExtractTable.vb
Imports Spire.Pdf
Imports Spire.Pdf.Utilities
Imports System.Text
Imports System.IO

Namespace ExtractTable
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Specify the input and output file paths
            Dim input As String = "..\..\..\..\..\..\Data\ExtractTable.pdf"
            Dim output As String = "ExtractTable_result.txt"

            ' Create a new PdfDocument object
            Dim pdf As New PdfDocument()

            ' Load the PDF document from the input file
            pdf.LoadFromFile(input)

            ' Create a PdfTableExtractor object for extracting tables from the document
            Dim extractor As New PdfTableExtractor(pdf)

            ' Initialize an array to store the extracted tables
            Dim tableLists() As PdfTable = Nothing

            ' Create a StringBuilder object to build the extracted text
            Dim builder As New StringBuilder()

            ' Iterate over each page of the document
            For pageIndex As Integer = 0 To pdf.Pages.Count - 1
                ' Extract tables from the current page
                tableLists = extractor.ExtractTable(pageIndex)

                ' Check if any tables were extracted
                If tableLists IsNot Nothing AndAlso tableLists.Length > 0 Then
                    ' Iterate over each extracted table
                    For Each table As PdfTable In tableLists
                        ' Get the number of rows and columns in the table
                        Dim row As Integer = table.GetRowCount()
                        Dim column As Integer = table.GetColumnCount()

                        ' Iterate over each cell in the table
                        For i As Integer = 0 To row - 1
                            For j As Integer = 0 To column - 1
                                ' Extract the text from each cell
                                Dim text As String = table.GetText(i, j)

                                ' Append the extracted text to the StringBuilder
                                builder.Append(text & " ")
                            Next j

                            ' Add a line break after each row
                            builder.Append(vbCrLf)
                        Next i
                    Next table
                End If
            Next pageIndex

            ' Write the extracted text to the output file
            File.WriteAllText(output, builder.ToString())

            ' Close the PdfDocument
            pdf.Close()

            ' Launch the Pdf files
            FileViewer(output)
		End Sub
		Private Sub FileViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/05_Tables/ExtractTable.cs
using System;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Utilities;
using System.Text;
using System.IO;

namespace ExtractTable
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Input and output file paths
            string input = @"..\..\..\..\..\..\Data\ExtractTable.pdf";
            string output = "ExtractTable_result.txt";

            //Create a PdfDocument
            PdfDocument pdf = new PdfDocument();

            //Load the input file from disk
            pdf.LoadFromFile(input);

            //Create PdfTableExtractor object to extract tables from PDF
            PdfTableExtractor extractor = new PdfTableExtractor(pdf);
            PdfTable[] tableLists = null;
            StringBuilder builder = new StringBuilder();

            //Iterate each page of PDF file
            for (int pageIndex = 0; pageIndex < pdf.Pages.Count; pageIndex++)
            {
                //Extract tables in each page
                tableLists = extractor.ExtractTable(pageIndex);

                if (tableLists != null && tableLists.Length > 0)
                {
                    //Iterate each table from array
                    foreach (PdfTable table in tableLists)
                    {
                        int row = table.GetRowCount();
                        int column = table.GetColumnCount();
                        for (int i = 0; i < row; i++)
                        {
                            for (int j = 0; j < column; j++)
                            {
                                //Extract the text in each cell
                                string text = table.GetText(i, j);

                                //Append the text into StringBuilder
                                builder.Append(text + " ");
                            }

                            builder.Append("\r\n");
                        }
                    }
                }
            }
            //Write all text 
            File.WriteAllText(output, builder.ToString());

            //Launch the Pdf files
            FileViewer(output);
        }
        private void FileViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}
